src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -748,7 +751,7 @@ private void fireCompleted(){
                     LOGGER.log(Level.WARNING, "Nulling out FlowExecution due to error in build " + this, x);
                     execution = null; // probably too broken to use
                     executionLoaded = true;
-                    saveWithoutFailing(); // Ensure we do not try to load again
+                    saveWithoutFailing(true); // Ensure we do not try to load again
```

Comment:
Should we call `completeAsynchronousExecution()` here?
```suggestion
                    saveWithoutFailing(true); // Ensure we do not try to load again
                    completeAsynchronousExecution();
```